src/number_theory/number_field/embeddings.lean

```diff
@@ -123,3 +124,170 @@ end
+by simp only [place, function.comp_app, map_zero, norm_zero]
+
+@[simp]
+lemma places.map_one : place φ 1 = 1 :=
+by simp only [place, function.comp_app, map_one, norm_one]
+
+@[simp]
+lemma places.map_inv (x : K) : place φ (x⁻¹) = (place φ x)⁻¹ :=
+by simp only [place, function.comp_app, norm_inv, map_inv₀]
+
+@[simp]
+lemma places.map_mul (x y : K) : place φ (x * y) = (place φ x) * (place φ y) :=
+by simp only [place, function.comp_app, map_mul, norm_mul]
+
+lemma places.add_le (x y : K) : place φ (x + y) ≤ (place φ x) + (place φ y) :=
+by simpa only [place, function.comp_app, map_add] using norm_add_le _ _
+
+end place
+
+end number_field
+
+namespace number_field.complex_embeddings
+
+open complex number_field
+
+open_locale complex_conjugate
+
+variables {K : Type*} [field K]
+
+/-- The conjugate of a complex embedding as a complex embedding. -/
+def conjugate (φ : K →+* ℂ) : K →+* ℂ := ring_hom.comp conj_ae.to_ring_equiv.to_ring_hom φ
+
+lemma conjugate_coe_eq (φ : K →+* ℂ) (x : K) : (conjugate φ) x = conj (φ x) := rfl
+
+lemma place_conjugate_eq_place (φ : K →+* ℂ) : place (conjugate φ) = place φ :=
+by { ext1, simp only [place, conjugate_coe_eq, function.comp_app, norm_eq_abs, abs_conj] }
+
+/-- A embedding into `ℂ` is real if it is fixed by complex conjugation. -/
+def is_real (φ : K →+* ℂ): Prop := conjugate φ = φ
+
+/-- A real embedding as a ring homomorphism from `K` to `ℝ` . -/
+def real_embedding {φ : K →+* ℂ} (hφ : is_real φ) : K →+* ℝ :=
+{ to_fun := λ x, (φ x).re,
+  map_one' := by simp only [map_one, one_re],
+  map_mul' := by simp only [complex.eq_conj_iff_im.mp (ring_hom.congr_fun hφ _), map_mul, mul_re,
+  mul_zero, tsub_zero, eq_self_iff_true, forall_const],
+  map_zero' := by simp only [map_zero, zero_re],
+  map_add' := by simp only [map_add, add_re, eq_self_iff_true, forall_const], }
+
+lemma real_embedding_eq_embedding {φ : K →+* ℂ} (hφ : is_real φ) (x : K) :
+  (real_embedding hφ x : ℂ) = φ x :=
+begin
+  ext, { refl, },
+  { rw [of_real_im, eq_comm, ← complex.eq_conj_iff_im],
+    rw is_real at hφ,
+    exact ring_hom.congr_fun hφ x, },
+end
+
+lemma place_real_embedding_eq_place {φ : K →+* ℂ} (hφ : is_real φ) :
+  place (real_embedding hφ) = place φ :=
+by { ext x, simp only [place, function.comp_apply, complex.norm_eq_abs, real.norm_eq_abs,
+  ← real_embedding_eq_embedding hφ x, abs_of_real] }
+
+lemma conjugate_conjugate_eq (φ : K →+* ℂ) :
+  conjugate (conjugate φ) = φ :=
+  by { ext1, simp only [conjugate_coe_eq, function.comp_app, star_ring_end_self_apply], }
+
+lemma conjugate_is_real_iff {φ : K →+* ℂ} :
+  is_real (conjugate φ) ↔ is_real φ := by simp only [is_real, conjugate_conjugate_eq, eq_comm]
+
+end number_field.complex_embeddings
+
+section infinite_places
+
+open number_field
+
+variables (K : Type*) [field K]
+
+/-- An infinite place of a number field `K` is a place associated to a complex embedding. -/
+def number_field.infinite_places := set.range (λ φ : K →+* ℂ, place φ)
```

Comment:
Maybe it's better as a `finset/fintype`?

Reply:
Yes, I was thinking about `infinite_place` being a `subtype`, that is 
```lean
def number_field.infinite_places := { w : K → ℝ // ∃ φ : K →+* ℂ, place φ = w }
```
(and the nice thing is that all the proofs stay the same :)
The fact that it is a `fintype` and computations of cardinalities is in the next PR #17753